components/timezone/src/windows_tz.rs

```diff
@@ -0,0 +1,202 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_provider::{
+    prelude::icu_locale_core::subtags::Region, DataError, DataPayload, DataProvider,
+};
+
+use crate::{
+    provider::windows::{WindowsZonesToBcp47MapV1, WindowsZonesToBcp47MapV1Marker},
+    TimeZoneBcp47Id,
+};
+
+/// A mapper between Windows time zone identifier and a BCP-47 ID.
+///
+/// This mapper currently only supports mapping from windows time zone identifiers
+/// to BCP-47 identifiers.
+///
+/// A windows time zone may vary depending on an associated territory/region. This is represented
+/// by the internal data mapping by delimiting the windows time zone and territory/region
+/// code with a "/".
+///
+/// For instance, Central Standard Time can vary depending on the provided regions listed below:
+///
+/// - Central Standard Time/001
+/// - Central Standard Time/US
+/// - Central Standard Time/CA
+/// - Central Standard Time/MX
+/// - Central Standard Time/ZZ
+///
+/// As such, a [`Region`] may be provided to further specify a desired territory/region when
+/// querying a BCP-47 identifier. If no region is provided or the specificity is not required,
+/// then the territory will default to the M49 World Code, 001.
+#[derive(Debug)]
+pub struct WindowsTimeZoneMapper {
+    data: DataPayload<WindowsZonesToBcp47MapV1Marker>,
+}
+
+#[cfg(feature = "compiled_data")]
+impl Default for WindowsTimeZoneMapper {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl WindowsTimeZoneMapper {
+    /// Creates a new [`WindowsTimeZoneMapper`].
+    #[cfg(feature = "compiled_data")]
+    pub fn new() -> Self {
```

Comment:
From https://github.com/unicode-org/icu4x/issues/5440, `WindowsTimeZoneMapper::new()` should return the borrowed type.